### ISSUE TITLE
Populating references on subdocuments in arrays that are sometimes undefined

### ISSUE THREAD

**Issue by Ullidtz** [Automattic/mongoose#7275]
**Do you want to request a *feature* or report a *bug*?**
Report a bug i believe (I'm guessing the behaviour is not by design)

**What is the current behavior?**
If I try to populate a reference on a subdocument in an array and that array is undefined an error will occur inside mpath and crash mongoose.

**If the current behavior is a bug, please provide the steps to reproduce.**
I believe this error should occur for any subdocuments in array with a reference being populated and hitting a document wherein the array containing the subdocuments is undefined.

Schema
```javascript
const wheelSchema = new Schema({
  type: { type: Schema.Types.ObjectId, ref: 'WheelType', required: true }
});

const carSchema = new Schema({
  wheels: [wheelSchema]
});

module.exports = new Schema({
  cars: [carSchema]
});
```

Query
```javascript
const carLot = await this.model()
      .findById(id)
      .populate({
          path: `cars.wheels.type`,
          model: 'InputType'
      });
```

Now if any of the lots have no cars or any of the cars have no wheels and no empty array as a placeholder for the wheels then the query population will fail, hard.

Stacktrace
```
TypeError: Cannot read property 'wheels' of undefined
    at Object.exports.get (secretpath/node_modules/mpath/lib/index.js:71:59)
    at search (secretpath/node_modules/mongoose/lib/helpers/populate/getSchemaTypes.js:110:23)
    at search (secretpath/node_modules/mongoose/lib/helpers/populate/getSchemaTypes.js:107:21)
    at getSchemaTypes (secretpath/node_modules/mongoose/lib/helpers/populate/getSchemaTypes.js:137:10)
    at getModelsMapForPopulate (secretpath/node_modules/mongoose/lib/model.js:3881:14)
    at populate (secretpath/node_modules/mongoose/lib/model.js:3486:21)
    at _populate (secretpath/node_modules/mongoose/lib/model.js:3456:5)
    at utils.promiseOrCallback.cb (secretpath/node_modules/mongoose/lib/model.js:3429:5)
    at Object.promiseOrCallback (secretpath/node_modules/mongoose/lib/utils.js:223:14)
    at Function.Model.populate (secretpath/node_modules/mongoose/lib/model.js:3428:16)
    at cb (secretpath/node_modules/mongoose/lib/query.js:1606:17)
    at result (secretpath/node_modules/mongodb/lib/utils.js:414:17)
    at executeCallback (secretpath/node_modules/mongodb/lib/utils.js:406:9)
    at handleCallback (secretpath/node_modules/mongodb/lib/utils.js:128:55)
    at cursor.close (secretpath/node_modules/mongodb/lib/operations/cursor_ops.js:218:62)
    at handleCallback (secretpath/node_modules/mongodb/lib/utils.js:128:55)
    at completeClose (secretpath/node_modules/mongodb/lib/cursor.js:887:14)
    at _endSession (secretpath/node_modules/mongodb/lib/cursor.js:898:37)
    at Cursor._endSession (secretpath/node_modules/mongodb-core/lib/cursor.js:193:5)
    at Cursor._endSession (secretpath/node_modules/mongodb/lib/cursor.js:226:59)
    at Cursor.close (secretpath/node_modules/mongodb/lib/cursor.js:898:19)
    at cursor._next (secretpath/node_modules/mongodb/lib/operations/cursor_ops.js:218:23)
    at handleCallback (secretpath/node_modules/mongodb-core/lib/cursor.js:202:5)
    at _setCursorNotifiedImpl (secretpath/node_modules/mongodb-core/lib/cursor.js:560:38)
    at self._endSession (secretpath/node_modules/mongodb-core/lib/cursor.js:568:46)
    at ClientSession.endSession (secretpath/node_modules/mongodb-core/lib/sessions.js:114:41)
    at Cursor._endSession (secretpath/node_modules/mongodb-core/lib/cursor.js:188:13)
    at Cursor._endSession (secretpath/node_modules/mongodb/lib/cursor.js:226:59)
    at _setCursorNotifiedImpl (secretpath/node_modules/mongodb-core/lib/cursor.js:568:17)
    at setCursorNotified (secretpath/node_modules/mongodb-core/lib/cursor.js:560:3)
    at setCursorDeadAndNotified (secretpath/node_modules/mongodb-core/lib/cursor.js:553:3)
    at nextFunction (secretpath/node_modules/mongodb-core/lib/cursor.js:783:5)
    at Cursor.next (secretpath/node_modules/mongodb-core/lib/cursor.js:824:3)
    at Cursor._next (secretpath/node_modules/mongodb/lib/cursor.js:211:36)
    at fetchDocs (secretpath/node_modules/mongodb/lib/operations/cursor_ops.js:211:12)
    at cursor._next (secretpath/node_modules/mongodb/lib/operations/cursor_ops.js:237:7)
    at handleCallback (secretpath/node_modules/mongodb-core/lib/cursor.js:202:5)
    at nextFunction (secretpath/node_modules/mongodb-core/lib/cursor.js:814:5)
    at secretpath/node_modules/mongodb-core/lib/cursor.js:690:7
    at queryCallback (secretpath/node_modules/mongodb-core/lib/cursor.js:267:16)
    at secretpath/node_modules/mongodb-core/lib/connection/pool.js:531:18
    at process._tickCallback (internal/process/next_tick.js:61:11)
[nodemon] app crashed - waiting for file changes before starting...
```

From what i can tell the error occurs inside mpath:
```javascript
      obj = _from instanceof Map ?
        _from.get(part) :
        _from[part];
````
An ugly hack that seems to work for my case would be to just check for falsy here:
```javascript
      obj = _from
        ? _from instanceof Map
          ? _from.get(part)
          : _from[part]
        : null;
```

But if we focus on what could be done inside mongoose then the issue I am running into is inside getSchemaTypes:
```javascript
              ret = search(parts.slice(p), foundschema.schema, mpath.get(trypath, subdoc));
```

Wrapping the mpath.get call in a try catch would work for me, but i suspect you guys may have a much better solution in mind.
```javascript
              let path = null;
              try {
                path = mpath.get(trypath, subdoc);
              } catch (e) {}
              ret = search(parts.slice(p), foundschema.schema, path); 
```

I hope this gives you everything you need.

**What is the expected behavior?**
To populate the references when they are present and not crash if an array is undefined.

**Please mention your node.js, mongoose and MongoDB version.**
node.js: 10.9.0
mongoose: 5.2.5 - (References mpath: 0.4.1)
MongoDB: 3.1.1

**Comment by vkarpov15** [comment]
Thanks for reporting, will look into this ASAP

**Comment by vkarpov15** [comment]
Turns out this is a duplicate of #6848 and #6845, please upgrade to [`>= 5.2.8` and this will work](https://github.com/Automattic/mongoose/blob/master/History.md#528--2018-08-13)
